### ISSUE TITLE
hadolint doesn't build for openSUSE Tumbleweed and Leap

### ISSUE THREAD

**Issue by SchoolGuy** [hadolint/hadolint#941]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

`hadolint` builds successfully in the openSUSE Build Service.

### Actual behavior

Builds crashes with

```
[   64s] /usr/bin/ghc --make -fbuilding-cabal-package -O -static -outputdir dist/build/hadolint/hadolint-tmp -odir dist/build/hadolint/hadolint-tmp -hidir dist/build/hadolint/hadolint-tmp -stubdir dist/build/hadolint/hadolint-tmp -i -idist/build/hadolint/hadolint-tmp -iapp -idist/build/hadolint/autogen -idist/build/global-autogen -Idist/build/hadolint/autogen -Idist/build/global-autogen -Idist/build/hadolint/hadolint-tmp -optP-include -optPdist/build/hadolint/autogen/cabal_macros.h -optl-static -optl-pthread -hide-all-packages -Wmissing-home-modules -no-user-package-db -package-db dist/package.conf.inplace -package-id base-4.14.3.0 -package-id containers-0.6.5.1 -package-id data-default-0.7.1.1-BL6m2BngHMsBEQPEliYgNY -package-id hadolint-2.12.1-GdMnfqVhALf1nfive1CJ5y -package-id language-docker-12.1.0-BCrNtoNHclute185TMqbn -package-id megaparsec-9.2.1-9kOCloyeDxHBHoFnSXOVz3 -package-id optparse-applicative-0.17.0.0-8GfpoPBHtf3Hh9WYymeqN -package-id prettyprinter-1.7.1-9SYNvhPK1YH2WAvQ3G3gEB -package-id text-1.2.4.1 -XHaskell2010 -XDeriveAnyClass -XDeriveGeneric -XNamedFieldPuns -XOverloadedStrings -XPatternSynonyms -XRecordWildCards -XScopedTypeVariables -XStrictData -XTemplateHaskell -XTupleSections app/Main.hs Paths_hadolint -o dist/build/hadolint/hadolint -Wall -Wcompat -Wincomplete-record-updates -Wincomplete-uni-patterns -Wredundant-constraints -optP-Wno-nonportable-include-path -flate-dmd-anal -O2 -threaded -rtsopts '-with-rtsopts=-N5 -A4m'
[   64s] Linking dist/build/hadolint/hadolint ...
[   64s] /usr/bin/ld.bfd: cannot find -lgmp: No such file or directory
[   64s] /usr/bin/ld.bfd: cannot find -lc: No such file or directory
[   64s] /usr/bin/ld.bfd: cannot find -lm: No such file or directory
[   64s] /usr/bin/ld.bfd: cannot find -lm: No such file or directory
[   64s] /usr/bin/ld.bfd: cannot find -lffi: No such file or directory
[   64s] /usr/bin/ld.bfd: cannot find -lnuma: No such file or directory
[   64s] /usr/bin/ld.bfd: cannot find -lc: No such file or directory
[   64s] collect2: error: ld returned 1 exit status
[   64s] `cc' failed in phase `Linker'. (Exit code: 1)
[   64s] error: Bad exit status from /var/tmp/rpm-tmp.RAPBc0 (%build)
```

### Steps to reproduce the behavior

1. `osc co home:SchoolGuy:ghc hadolint`
2. `osc build`

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Not yet executable
```

**Dockerfile** (if relevant)

```Dockerfile
None used
```

**Additional environment details (OS, stack version, etc.)**

openSUSE Build Service package: https://build.opensuse.org/package/show/home:SchoolGuy:ghc/hadolint

The specfile was created with <https://github.com/peti/cabal2spec>

I also created an issue in the openSUSE Bugzilla since it might be related to how ghc is built in the OBS however, my Haskell knowledge is zero sadly.

During the build process in the openSUSE Build Service the worker doesn't have any internet connection and only source files can be used in combination with build dependencies. Fancy things like ghcup won't work sadly to solve this.

Edit: Specfile for current version of ghc can be found here: <https://build.opensuse.org/package/view_file/devel:languages:haskell/ghc/ghc.spec?expand=1>

**Comment by SchoolGuy** [comment]
Adding to my incompetence I tried to create a reproducer... It seems I am missing some key Haskell knowledge or something is very weird...

```
# podman run --rm -it registry.opensuse.org/opensuse/tumbleweed
# zypper in -y ghc cabal-install git
# git clone https://github.com/hadolint/hadolint \
   && cd hadolint \
   && git checkout v2.12.0 \
   && cabal update \
   && cabal configure \
   && cabal build all \
   && cabal install
```

This does nothing except produce the following error:

```
cabal: Unknown target '.'.
There is no component '.'.
The project has no package directory '.'.
```

If I specifically ask for hadolint it tells me the following:

```
476c58ed3e0b:/hadolint # cabal install exe:hadolint
cabal: Internal error in target matching. It should always be possible to find
a syntax that's sufficiently qualified to give an unambiguous match. However
when matching 'exe:hadolint' we found exe:hadolint (unknown-component) which
does not have an unambiguous syntax. The possible syntax and the targets they
match are as follows:
'exe:hadolint' which matches exe:hadolint (unknown-component),
:pkg:exe:lib:exe:file:hadolint (unknown-file)
```

When trying to follow the hint from [the Cabal Install online help](https://cabal.readthedocs.io/en/stable/cabal-commands.html#cabal-install) something is happening but not what I would expect.

```
cabal install all:exes
```

Which weirdly results in Shellcheck being installed:

```
476c58ed3e0b:/hadolint # ls -alh /root/.cabal/bin
total 4.0K
drwxr-xr-x 1 root root  20 Mar  1 06:41 .
drwxr-xr-x 1 root root  52 Mar  1 06:41 ..
lrwxrwxrwx 1 root root 116 Mar  1 06:41 shellcheck -> ../store/ghc-8.10.7/ShellCheck-0.8.0-d976271963061fb0959444b00bbbad98671b6036f8fa74071346e7849678cbe3/bin/shellcheck
```

**Comment by SchoolGuy** [comment]
I found my issue: In the OBS we have a control file called `_service` which automatically downloads the source tarball. I set the version correctly however I forgot to set the revision which resulted in the fact that I used git HEAD instead of the desired 2.12.0. That led to different dependencies than required for 2.12.0 which then lead to my build errors.

For context: version and revision are two different parameters in our context because maybe you want the version downstream to stay at a desired place and take one or two fixup commits that are past certain tags. Doesn't make sense in most cases but for downstream Linux packaging this is a useful mechanism. Which in this case bit me... 

Thanks to the effort of my colleagues we now have `hadolint` in openSUSE: https://build.opensuse.org/package/show/devel:languages:haskell:ghc-9.4.x/hadolint
